src/main/scala/io/iohk/ethereum/utils/Picklers.scala

```diff
@@ -30,9 +33,18 @@ object Picklers {
 
   implicit val addressPickler: Pickler[Address] =
     transformPickler[Address, ByteString](bytes => Address(bytes))(address => address.bytes)
-  implicit val transactionPickler: Pickler[LegacyTransaction] = generatePickler[LegacyTransaction]
+  implicit val accessListItemPickler: Pickler[AccessListItem] = generatePickler[AccessListItem]
+
+  implicit val legacyTransactionPickler: Pickler[LegacyTransaction] = generatePickler[LegacyTransaction]
+  implicit val transactionWithAccessListPickler: Pickler[TransactionWithAccessList] =
+    generatePickler[TransactionWithAccessList]
+
+  implicit val transactionPickler: Pickler[Transaction] = compositePickler[Transaction]
+    .addConcreteType[LegacyTransaction]
+    .addConcreteType[TransactionWithAccessList]
+
   implicit val signedTransactionPickler: Pickler[SignedTransaction] =
-    transformPickler[SignedTransaction, (LegacyTransaction, ECDSASignature)] { case (tx, signature) =>
+    transformPickler[SignedTransaction, (Transaction, ECDSASignature)] { case (tx, signature) =>
```

Comment:
note: Won't we have to nuke the storage because the block body are now incompatible ?

Reply:
yep, this will indeed create an incompatibility. I think we can also re-introduce the chainId change and release then.
as long as we can stagger the release to sagano - we will not loose data, but this will require some devops-ing.